packages/itwinui-react/src/core/Menu/MenuItem.tsx

```diff
@@ -126,17 +147,27 @@ export const MenuItem = React.forwardRef((props, ref) => {
       case 'ArrowRight': {
         if (subMenuItems.length > 0) {
           setIsSubmenuVisible(true);
+
+          // flush and reset state so we are ready to focus again next time
+          flushSync(() => setFocusOnSubmenu(true));
+          setFocusOnSubmenu(false);
```

Comment:
after removing `handleFocusOut`, i'm having a bit of difficulty keeping the menus open. when ArrowRight is pressed, focus moves into the submenu and closes the parent menu.

Reply:
can we remove hover/focus behavior and instead use clicks?

we can always explore adding it back in the future if needed? it would also solve #879.

Reply:
I agree with using clicks.
Maybe switching to controlled mode would solve this? Like parent visible:true once you click on a submenu?

Reply:
it's already using controlled mode, but there are many conditions under which we want to set visible:false, one of which is when focus leaves the menu.

i'm looking into changing the dom structure (currently the submenu is a sibling of the parent menu as they get portaled separately).

Reply:
fixed by removing `FloatingFocusManager` and adding a simple `onKeyDown` listener on `Menu`
<img width="307" alt="" src="https://github.com/iTwin/iTwinUI/assets/9084735/c77c1019-7bb9-4e28-bf08-adb094d2bea4">

this works because:
- `close()` will send focus back to the trigger, thus preserving original tab order and avoiding the need for complex focus management.
- events inside react portals bubble to the react parents, so `Menu` will get keydown events from submenus even if they are sublings in the dom.